### PR TITLE
HttpProvider: CORS issue with Firefox and Safari

### LINKED ISSUE #2978: CORS issue with 1.2.0 on Firefox and Safari
Steps to reproduce:

 * Import web3 1.2.0
 * `var web3 = new Web3(new Web3.providers.HttpProvider('https://rinkeby.infura.io/'));`
 * `web3.eth.getTransaction('0xd540f813d46953f5bf7c0b3797572ff6b73eee0eeadfac071baee0da9a7a9b5a').then(console.log);`

You will get the following error messages:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://rinkeby.infura.io/. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://rinkeby.infura.io/. (Reason: CORS request did not succeed).
Error: Invalid JSON RPC response: ""
```

Doesn't happen on Brave or Chrome or the new Edge, but happens on Firefox, Safari and Safari Mobile.

### PR DESCRIPTION
**Pull request by nivida**
Fixes #2978 and adds the options property ``withCredentials``.

### CODE REVIEW

**Comment by nivida** [comment]
Sorry for the confusion with closing and opening of the PR GitHub had a lag.

**Comment by nivida** [comment]
Added the in-progress label again because it can't find the global ``process`` var with nodejs v12.

File: dist/web3.min.js [hunk 1]
```diff
@@ -1 +0,0 @@
...,e,r,n,i){if(B(t,Uint8Array)&&(t=l.from(t,t.offset,t.byteLength)),!l.isBuffer(t))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+_typeof(t));if(void 0===e&&(e=0),void 0===r&&(r=t?t.length:0),void 0===n&&(n=0),void 0===i&&(i=this.length),e<0||r>t.length||n<0||i>this.length)throw new RangeError("out of range index");if(i<=n&&r<=e)return 0;if(i<=n)return-1;if(r<=e)return 1;if(this===t)return 0;for(var o=(i>>>=0)-(n>>>=0),a=(r>>>=0)-(e>>>=0),u=Math.min(o,a),s=this.slice(n,i),c=t.slice(e,r),f=0;f<u;++f)if(s[f]!==c[f]){o=s[f],a=c[f];break}return o<a?-1:a<o?1:0},l.prototype.includes=function(t,e,r){return-1!==this.indexOf(t,e,r)},l.prototype.indexOf=function(t,e,r){return p(this,t,e,r,!0)},l.prototype.lastIndexOf=function(t,e,r){return p(this,t,e,r,!1)},l.prototype.write=function(t,e,r,n){if(void 0===e)n="utf8",r=this.length,e=0;else if(void 0===r&&"string"==typeof e)n=e,r=this.length,e=0;else{if(!isFinite(e))throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");e>>>=0,isFinite(r)?(r>>>=0,void 0===n&&(n="utf8")):(n=r,r=void 0)}var i=this.length-e;if((void 0===r||i<r)&&(r=i),0<t.length&&(r<0||e<0)||e>
```
Comment (cgewecke) [inline dist/web3.min.js:1]: Is this removal related to the fix?

Comment (nivida) [reply dist/web3.min.js:1]: Yes, it is because some devs will use the minified file from the working branch (1.x). :)

**Review by cgewecke — APPROVED** [review]
This LGTM - it works in 2.x with no problems?  

It would be cool to try to reproduce the CORS error (at least for Firefox) in headless browser testing but those fixtures need to be built.

What about docs changes for the new option?

**Comment by nivida** [comment]
@cgewecke Yeah, it's really strange and couldn’t clearly figure out why the process global var isn't existing in nodejs v12 for these tests. 

> It would be cool to try to reproduce the CORS error (at least for Firefox) in headless browser testing but those fixtures need to be built.

I wrote it down on my todo list and will mention it on the missing comment to your e2e issue.

> What about docs changes for the new option?

Providers weren’t documented until now. I wrote that down on the todo list as well and will create a related issue for it today. 

I will merge this PR for now.
